apps/coscrad-frontend/src/utils/generic-components/presenters/detail-views/resource-preview-icon.tsx

```diff
@@ -68,5 +68,9 @@ export const ResourcePreviewIconFactory = ({
         throw new Error(`Failed to build an icon for resource type: ${resourceType}`);
     }
 
-    return <Box sx={{ fontSize: iconSize, maxHeight: iconSize, color: color }}>{lookupResult}</Box>;
+    return (
+        <Box sx={{ fontSize: iconSize, maxHeight: iconSize, color: color }}>
+            <Tooltip title={resourceType}>{lookupResult}</Tooltip>
```

Comment:
nice!

Reply:
But do note that we shouldn't be showing the `resourceType` to the user. We need to show a label. I don't have enough context here. It's possible that we want to show the label from the config  for this particular index-to-detail flow. Remember that there can be multiple configured index-to-detail flows for a single resource type. I won't block this PR, but we should look at this detail in the near-future.